### PR TITLE
use elapsed macro rather than timing for loop, also allow seconds

### PR DESCRIPTION
**Pull request by hayd**
This seems like it may be "fairer", though I'm not sure if it is more accurate or not. Thought I should throw it out there.

_There is also printing in seconds (if it takes that long)._

### CODE REVIEW

File: src/TimeIt.jl [hunk 1]
```diff
@@ -4,12 +4,12 @@ export @timeit
 
 macro timen(ex, n)
     quote
-        local t0 = time_ns()
+        local total = 0.0
         for i = 1:$(esc(n))
-            local val = $(esc(ex))
+            # already warm
+            total += @elapsed $(esc(ex))
```
Comment (hayd) [inline src/TimeIt.jl:10]: Ag. in retrospect this is obviously not going to be as accurate (as there is an accuracy limit which is avoided using the for  loop). Saying that, we can still use `@elapsed` here (but over the for loop), will update. One point to mention is the type inference of val - that could slows things down. :s

Comment (kbarbary) [inline src/TimeIt.jl:10]: Thanks for the PR! I'm actually traveling for the next week and a half, but
I'll be sure to have a look at this when I'm back.

Regarding type inference on "val", I'm most worried about the whole macro
being in global scope. I'm not sure I did that right. Might be better to
have the macro define a function and immediately call it.

**Comment by kbarbary** [comment]
Perhaps instead of having the separate `@timen` macro, the `@timeit` macro should define a function, say, `testloop(n)` (or whatever), that includes the loop and the input expression, and takes a single argument `n` (the number of loop iterations). Then we could just use `@elapsed testloop(n)` in place of `@timen` calls. 

Updated PR welcome, otherwise I might just do it myself... eventually.

**Comment by hayd** [comment]
when I tried this, I seemed to sometimes get the calling of the function compiled away! I don't know if this was because some simple functions can be inlined and then are seen to have no effect... Will try again sometime :)

**Comment by kbarbary** [comment]
I've heard of this happening. Were you returning a result in the function? If the expression value isn't used, it might be that the compiler optimizes it out.

**Pull request closed without merging**